src/wallet/rpcwallet.cpp

```diff
@@ -3094,6 +3129,18 @@ void FundTransaction(CWallet* const pwallet, CMutableTransaction& tx, CAmount& f
         setSubtractFeeFromOutputs.insert(pos);
     }
 
+    // Fetch specified UTXOs from the UTXO set
```

Comment:
Not immediately obvious to me why this change is necessary. 

Probably requires more extensive comment?

Reply:
In other words, how did selection of inputs work before this?

Reply:
CTxOuts are needed for fee estimation and knowing the values of the outputs being selected. For inputs the wallet knows about, it would fetch the outputs from the wallet (obviously). Otherwise it would basically ignore them. So to support external inputs, either the user would have to enter the output info (bad UX IMO) or we can fetch it from the UTXO set.

Reply:
oh, duh pubkeys/descriptors aren't enough :)

Suggested text:
"// Fetch specified UTXOs from the UTXO set for knowing the values of the outputs being selected
// and to match with the given solving_data. These are only
// used when the outputs are not stored in the wallet."

Reply:
Done